File: app/src/main/java/mx/edu/upqroo/kristenandroid/ui/fragments/CalendarFragment.java
package mx.edu.upqroo.kristenandroid.ui.fragments;


import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebChromeClient;
import android.webkit.WebView;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;
import org.jetbrains.annotations.NotNull;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import mx.edu.upqroo.kristenandroid.R;
import mx.edu.upqroo.kristenandroid.api.kristen.messages.CalendarUrlMessage;

/**
 * A simple {@link Fragment} subclass.
 */
public class CalendarFragment extends Fragment {
    private WebView mWebView;

    public CalendarFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @SuppressLint("SetJavaScriptEnabled")
    @Override
    public View onCreateView(@NotNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View v = inflater.inflate(R.layout.fragment_calendar, container, false);
        mWebView = v.findViewById(R.id.webViewCalendar);
        mWebView.setWebChromeClient(new WebChromeClient());
        mWebView.getSettings().setJavaScriptEnabled(true);
        mWebView.getSettings().setBuiltInZoomControls(true);
        return v;
    }

    @Override
    public void onStart() {
        super.onStart();
        EventBus.getDefault().register(this);
    }

    @Override
    public void onStop() {
        super.onStop();
        EventBus.getDefault().unregister(this);
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void calendarServiceResponse(CalendarUrlMessage message) {
        if (!message.getCalendarUrl().isEmpty()) {
            mWebView.loadUrl("https://docs.google.com/gview?embedded=true&url="
                    + message.getCalendarUrl());
        }
    }

}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/ui/fragments/CalendarFragment.kt
package mx.edu.upqroo.kristenandroid.ui.fragments


import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebChromeClient
import android.webkit.WebView
import androidx.fragment.app.Fragment
import mx.edu.upqroo.kristenandroid.R
import mx.edu.upqroo.kristenandroid.api.kristen.messages.CalendarUrlMessage
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

/**
 * A simple [Fragment] subclass.
 */
class CalendarFragment : Fragment() {
    private lateinit var mWebView: WebView

    @SuppressLint("SetJavaScriptEnabled")
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val v = inflater.inflate(R.layout.fragment_calendar, container, false)
        mWebView = v.findViewById(R.id.webViewCalendar)
        mWebView.webChromeClient = WebChromeClient()
        mWebView.settings.javaScriptEnabled = true
        mWebView.settings.builtInZoomControls = true
        return v
    }

    override fun onStart() {
        super.onStart()
        EventBus.getDefault().register(this)
    }

    override fun onStop() {
        super.onStop()
        EventBus.getDefault().unregister(this)
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun calendarServiceResponse(message: CalendarUrlMessage) {
        if (message.calendarUrl.isNotEmpty()) {
            mWebView.loadUrl(
                    "https://docs.google.com/gview?embedded=true&url=" + message.calendarUrl)
        }
    }

}
